### PR TITLE
Fix API usage

### PR DESCRIPTION
**Pull request by jmoreira-valory**
Fix excessive API usage

### CODE REVIEW

File: packages/valory/skills/market_creation_manager_abci/behaviours.py [hunk 1]
```diff
@@ -2179,12 +2180,34 @@ def _parse_llm_output(
...
     def _get_answer(self, question: str) -> Generator[None, None, Optional[str]]:
         """Get an answer for the provided questions"""
-        markets_to_approve_per_epoch = self.params.markets_to_approve_per_epoch
-        self.context.logger.info(
-            f"markets_to_approve_per_epoch={markets_to_approve_per_epoch}"
-        )
+
+        # An initial query is made to Newsapi to detect ratelimit issue
+        # This query is also included in the input_news passed to the LLM,
+        # if the call succeeds.
```
Comment (0xArdi) [inline packages/valory/skills/market_creation_manager_abci/behaviours.py:2202]: where?

File: packages/valory/skills/market_creation_manager_abci/behaviours.py [hunk 2]
```diff
@@ -2179,12 +2180,34 @@ def _parse_llm_output(
...
+        input_news = ""
+        initial_news_articles = yield from self._get_news(question)
```
[the remark this answers is not in the document]
Comment (0xArdi) [reply packages/valory/skills/market_creation_manager_abci/behaviours.py:2204]: how does this actually solve the issue?

[the remark this answers is not in the document]
Comment (jmoreira-valory) [reply packages/valory/skills/market_creation_manager_abci/behaviours.py:2204]: The problem is that it first called OpenAI and then newsapi... if newsapi was ratelimited, the OpenAI call was wasted.
Also, the rewiring of the FSM solved useless continuous OpenAI calls, where the agent was stuck.

**Review by 0xArdi — APPROVED** [review]

File: packages/valory/skills/market_creation_manager_abci/behaviours.py [hunk 3]
```diff
@@ -2392,6 +2407,7 @@ def get_payload(self) -> Generator[None, None, str]:
                 self.context.logger.warning(
                     f"Couldn't get answer for question {question}"
                 )
+                ############
                 continue
```
Comment (Adamantios) [inline packages/valory/skills/market_creation_manager_abci/behaviours.py:2411]: Accidental?

**Pull request merged**